src/middleware/setUserContext.js

```diff
@@ -0,0 +1,16 @@
+const pool = require("../config/db");
+
+const setUserContext = async (req, res, next) => {
+  try {
+    const userId = req.user.sub;
+    console.log("This is User_id : "+userId);
+   const result = await pool.query("SELECT set_config('app.user_id',$1, false)", [userId]);
```

Comment:
Here, the 3rd argument should be `true`. Then this configuration is only applied to the current database transaction. Otherwise it applies to the session.

Reply:
Where have we enabled RLS for the `user` table? Would be good to have all the DDLs inside the repo as well.